File: app/src/main/java/com/sami/rippel/base/SimpleFragment.java
package com.sami.rippel.base;

import android.app.Activity;
import android.content.Context;
import android.os.Bundle;
import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

/**
 * Created by yassine on 11/10/17.
 */

public abstract class SimpleFragment extends Fragment {

    protected View rootView;
    protected Activity mActivity;
    protected Context mContext;

    public abstract int getFragmentId();

    @Override
    public void onAttach(Context context) {
        mActivity = (Activity) context;
        mContext = context;
        super.onAttach(context);
    }

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater,
                             @Nullable ViewGroup container,
                             @Nullable Bundle savedInstanceState) {
        rootView = inflater.inflate(getFragmentId(), null);
        return rootView;
    }
}

File: base-android/src/main/java/com/sfaxdroid/base/SimpleFragment.kt
package com.sfaxdroid.base

import android.app.Activity
import android.content.Context
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment

/**
 * Created by yassine on 11/10/17.
 */
abstract class SimpleFragment : Fragment() {
    @JvmField
    protected var rootView: View? = null
    protected var mActivity: Activity? = null

    abstract val fragmentId: Int

    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        rootView = inflater.inflate(fragmentId, null)
        return rootView
    }
}